src/jit/emitxarch.cpp

```diff
@@ -4512,10 +4512,13 @@ void emitter::emitIns_R_R_S_I(
 //    opReg encoded in imm[7:4]
 static int encodeXmmRegAsIval(regNumber opReg)
 {
-    assert(opReg >= XMMBASE);
     // AVX/AVX2 supports 4-reg format for vblendvps/vblendvpd/vpblendvb,
     // which encodes the fourth register into imm8[7:4]
-    return (opReg - XMMBASE) << 4;
+    assert(opReg >= XMMBASE);
+    int ival = (opReg - XMMBASE) << 4;
+
+    assert((ival >= 0) && (ival <= 255));
```

Comment:
> But I would at least not cast to a signed type (int8_t) when you're using an unsigned range above.

@CarolEidt, we actually can't do that and both are required.

The JIT does signed check (value must be in range of -128 to 127) for "fits in byte". However, the value returned by the computation, (opReg - XMMBASE) << 4, is within the unsigned range (0 to 255).

If we cast to uint8_t, then the value is zero-extended, rather than sign-extended, and we will fail the "fits in byte" check. This was the same problem that was plaguing the HWIntrinsic APIs that directly take an immediate parameter (as they take a byte). -- I think we also discussed, briefly, that this may cause us to emit code that is larger than necessary in some cases (but I don't see an issue tracking that)...